crates/aptos-logger/src/aptos_logger.rs

```diff
@@ -1024,4 +1055,135 @@ mod tests {
                 "source_path"
             )));
     }
+
+    // Test speculative logging via SpeculativeEvents, here because the testing infrastructure can be re-used.
```

Comment:
FWIW, I think this fits better in the speculative crate. :) Unless there's a dependency in this test mod that I'm missing?

Reply:
moved!